src/main/java/org/isf/disease/service/DiseaseIoOperationRepository.java

```diff
@@ -31,7 +31,8 @@
 
 @Repository
 public interface DiseaseIoOperationRepository extends JpaRepository<Disease, String> {
-
+	
+	@Query(value = "select d FROM Disease d WHERE  d.code = :code")    // d.diseaseType.code = :code
```

Comment:
is this annotation really needed since it's performing the same query as per [JpaRepository query keywords](https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#repository-query-keywords)?